util/c_function.py

```diff
@@ -295,3 +295,31 @@ def get_summary_for_prompt(self) -> str:
             postconds_in_prompt = ", ".join(self.postconditions)
             function_name_and_signature += f"Postconditions: {postconds_in_prompt}\n"
         return function_name_and_signature
+
+    def __hash__(self) -> int:
+        """Return a hash based on this function's name, location, and file content.
+
+        Returns:
+            int: The hash based on this function's name, location, and file content.
+        """
+        self_file_content = pathlib.Path(self.file_name).read_text()
+        return hash((self.name, self_file_content, self.start_line, self.end_line))
+
+    def __eq__(self, other: object) -> bool:
+        """Return True iff this function is equivalent to another.
+
+        Returns:
+            bool: True iff this function is equivalent to another.
+        """
+        if not isinstance(other, CFunction):
+            return False
+        self_file_content = pathlib.Path(self.file_name).read_text()
+        other_file_content = pathlib.Path(other.file_name).read_text()
+        return (
+            # MDE: if the file names are the same, what is the point of comparing the content?
```

Comment:
@mernst If you are referring to the line immediately below (i.e., `self.name == other.name`), that is a comparison of _function_ names, not _file_ names.